python/cudf/cudf/_lib/copying.pyx

```diff
@@ -392,35 +375,31 @@ cdef class _CPackedColumns:
 
         header["column-names"] = self.column_names
         header["index-names"] = self.index_names
-        if self.c_obj.metadata.get()[0].data() != NULL:
+        cdef PlcPackedColumns p = self._data
+        if p.c_obj.get()[0].metadata.get()[0].data() != NULL:
             header["metadata"] = list(
-                <uint8_t[:self.c_obj.metadata.get()[0].size()]>
-                self.c_obj.metadata.get()[0].data()
+                <uint8_t[:p.c_obj.get()[0].metadata.get()[0].size()]>
+                p.c_obj.get()[0].metadata.get()[0].data()
```

Comment:
And I think we can simplify all of this to send `self._metadata` and `self._gpudata` directly.